File: src/main/java/io/github/dylmeadows/eontimer/model/resource/ImageResource.java
package io.github.dylmeadows.eontimer.model.resource;

import io.github.dylmeadows.common.core.io.Resource;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ImageResource implements Resource {
    DEFAULT_BACKGROUND_IMAGE("default_background_image.png");

    private final String path;
}

File: src/main/kotlin/io/github/dylmeadows/eontimer/model/resource/ImageResource.kt
package io.github.dylmeadows.eontimer.model.resource

import io.github.dylmeadows.common.core.io.Resource

enum class ImageResource(private val path: String) : Resource {
    DefaultBackgroundImage("default_background_image.png");

    override fun getPath(): String {
        return "$BASE_RESOURCE_PATH/img/${this.path}"
    }
}